doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
```

Comment:
Perhaps write
```
A *functional* is an operator :math:`f: X \to \mathbb{F}` that maps ...
```
without the displayed math line.
I'm also not sure I like the notation `f` for functional, I think it's too close to functions. My preferred notation would be something like `S(f)` for the functional value of `S` at `f`, since in 90 % of the cases, the input will be a function.